src/Invitation/InvitationForm.js

```diff
@@ -1,16 +1,120 @@
+    'http://tineye.com/images/widgets/mona.jpg',
+    'http://tineye.com/images/widgets/mona.jpg',
+  ]);
+  const galleryArray = [];
+  const hiddenMainPhotoInput = useRef(null);
+  const hiddenGalleryInput = useRef(null);
+
+  const uploadMainPhoto = () => {
+    hiddenMainPhotoInput.current.click();
+  };
+  const addPicToGallery = () => {
+    hiddenGalleryInput.current.click();
+  };
+
+  const previewMainPhoto = (e) => {
+    const reader = new FileReader();
+    reader.onload = () => {
+      if (reader.readyState === 2) {
+        setMainPhoto(reader.result);
+      }
+    };
+    reader.readAsDataURL(e.target.files[0]);
+  };
+
+  const updateMessage = (e) => {
+    setInviteMessage(e.target.value);
+  };
+
+  const previewGallery = (e) => {
+    const selectedPhotos = e.target.files;
+    for (let i = 0; i < selectedPhotos.length; i++) {
+      galleryArray.push(URL.createObjectURL(selectedPhotos[i]));
+    }
+    setGalleryURL([...galleryArray, ...galleryURL]);
+  };
+
   return (
-    <>
-      <h2>This is Invitaion Page</h2>
-      <Link to="/rsvp-preview">
-        <Button>Click me to Go to RSVP Page</Button>
-      </Link>
-    </>
+    <Container
+      style={{
+        width: '65%',
+        // backgroundImage: 'url(http://tineye.com/images/widgets/mona.jpg)',
+        // backgroundRepeat: 'no-repeat',
+        // height: '100%', ----------TO BE REVISED--------------
+      }}
+    >
+      <h2>Your Wedding Website</h2>
+      <Stack className="mb-5" direction="horizontal" gap={3}>
+        <PlumButton className="ms-auto">Share your website</PlumButton>
+        <Link to="/rsvp-preview" state={{ from: galleryURL }}>
+          <PlumFilledButton>Preview Site</PlumFilledButton>
+        </Link>
+      </Stack>
+      <div>Main Photo</div>
+      <img src={mainPhoto} alt="mainPhoto" width="100%" height="200" />
+      <input
+        type="file"
+        ref={hiddenMainPhotoInput}
+        accept="image/*"
+        style={{ display: 'none' }}
+        onChange={previewMainPhoto}
+      />
+      <div className="d-flex justify-content-center mt-2">
+        <PlumButton onClick={uploadMainPhoto}>Choose new photo</PlumButton>
+      </div>
+      <Form className="mt-3">
+        <Form.Group className="mb-3" controlId="formMessage">
+          <Form.Label>Message</Form.Label>
+          <Form.Control as="textarea" onChange={updateMessage} />
+        </Form.Group>
+        <Form.Group className="mb-3" controlId="formMessage">
+          <div>Gallery</div>
+          <CardGroup>
+            {(galleryURL || []).map((url) => (
+              <Card>
+                <Card.Img src={url} alt="gallery" variant="top" height="200" />
+              </Card>
+            ))}
+          </CardGroup>
+          <Form.Control
+            type="file"
+            ref={hiddenGalleryInput}
+            accept="image/*"
+            style={{ display: 'none' }}
+            onChange={previewGallery}
+            multiple
+          />
```

Comment:
In your next refactor, you may need to adjust the img attributes in order for the gallery photos to keep their aspect ratio but be clipped to fill the desired area for all screen sizes.